File: go/number-of-zero-filled-subarrays.go
func zeroFilledSubarray(nums []int) int64 {
	subarraySize := int64(0)
	subarrayCount := int64(0)
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			subarraySize += 1
		} else {
			subarraySize = 0
		}
		subarrayCount += subarraySize
	}
	return subarrayCount
}

File: python/number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        subarray_count = 0
        subarray_size = 0
        for num in nums:
            if num == 0:
                subarray_size += 1
            else:
                subarray_size = 0
            subarray_count += subarray_size
        return subarray_count
